src/error.rs

```diff
@@ -50,53 +52,40 @@ pub enum CryptoCoreError {
 impl Display for CryptoCoreError {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
-            Self::DeserializationEmptyError => {
-                write!(f, "empty input when parsing bytes")
-            }
-            Self::DeserializationSizeError { given, expected } => write!(
-                f,
-                "wrong size when parsing bytes: {given} given should be {expected}"
-            ),
-            #[cfg(feature = "ser")]
-            Self::ReadLeb128Error(err) => write!(f, "when reading LEB128, {err}"),
-            Self::GenericDeserializationError(err) => {
-                write!(f, "deserialization error: {err}")
-            }
-            Self::WriteLeb128Error { value, error } => {
-                write!(f, "when writing {value} as LEB128 size, IO error {error}")
-            }
-            Self::SerializationIoError { bytes_len, error } => {
-                write!(f, "when writing {bytes_len} bytes, {error}")
-            }
-            Self::PlaintextTooBigError { plaintext_len, max } => write!(
-                f,
-                "when encrypting, plaintext of {plaintext_len} bytes is too big, max is {max} \
-                 bytes"
-            ),
-            #[cfg(any(feature = "certificate", feature = "nist_curves"))]
-            Self::Pkcs8Error(err) => write!(f, "when converting to PKCS8, {err}"),
             #[cfg(any(feature = "certificate", feature = "nist_curves"))]
-            Self::Certificate(err) => write!(f, "when building certificate, {err}"),
-            Self::CiphertextTooSmallError {
+            CryptoCoreError::Certificate(err) => write!(f, "when building certificate, {err}"),
+            CryptoCoreError::CiphertextTooSmallError {
```

Comment:
Why this change?

Reply:
Because this is the recommended way of doing it

Reply:
I thought this was this opposite: https://rust-lang.github.io/rust-clippy/master/index.html#use_self

Reply:
When you let Rust analyzer generate match cases for Enums, it will ignore the Self ones and generate then with the fully qualified name. I believe rust analyzer uses clippy recommendations behind the hood. 
I honestly believe this is the correct choice for a match where you do not want ambiguity in the enumeration; the opposite would be true for a "constructor" which is what you example shows (even though there is little to none inheritance in Rust)